elastic/kibana
# [data visualizer] fix clicking field statistics 'Explore in maps' button does not take users to maps

Fixes https://github.com/elastic/kibana/issues/181900

PR resolves issue by
1. strongly typing `triggerOptions` to `VisualizeFieldContext` so future type changes to `VisualizeFieldContext` will be picked up by this file
2. Add more protection to `visualizeGeoFieldAction.isCompatible` method to prevent executing invalid context.

Out of scope for PR - updating "field statistics" to not show "Explore in maps" when action compatibility check fails. `visualizeGeoFieldAction.isCompatible` is async while `getActions` is sync.

### test
1. install sample web logs
4. Open discover
5. Change data table view to field statistics by clicking "Field statistics" toggle.
6. Click "Explore in maps" action button for field `geo.coordinates`.
7. Verify map is opened showing geo.coordinates
